### DISCUSSION TITLE
Type definition issue

### DISCUSSION THREAD

**Discussion by TheoMer** [prisma/prisma#3282] [Q&A] [answered]
So I have a field with the following type

```ts
schema.mutationType({
  definition(t) {
    t.field('createItem', {
      type: "Item",
      nullable: false,
      args: {
        itemvariants: schema.arg({ type: "ItemVariantsCreateManyWithoutItemInput", list: true, nullable: true }),
      },
      resolve: async (_, args, ctx) => {
        return await ctx.db.item.create(
          {
            data: {
              itemvariants: args.itemvariants,
            },
          },
        )
      }
    })
```

But `itemvariants: args.itemvariants` is throwing a conflict error because it says itemvariants is defined as

```
(property) itemvariants?: ItemVariantsCreateManyWithoutItemInput | null | undefined

Type '{ connect?: { id?: string | null | undefined; } | null | undefined; create?: { cartitems?: { connect?: { id?: string | null | undefined; } | null | undefined; create?: { createdAt?: any; id?: string | ... 1 more ... | undefined; quantity?: number | ... 1 more ... | undefined; updatedAt?: any; User: { ...; }; } | nul...' is not assignable to type 'ItemVariantsCreateManyWithoutItemInput | null | undefined'.
```

while index.d.ts is expecting:

```ts
itemvariants?: ItemVariantsCreateManyWithoutItemInput | null
```

How do I resolve this?

My model for Item is

```prisma
model Item {
  ...
  itemvariants    ItemVariants[]
}
```

**Comment by ryands17** [comment]
Hey @TheoMer 👋 
The Nexus InputObject `ItemVariantsCreateManyWithoutItemInput` should exactly match the Prisma type otherwise this error will occur. So do check if the types are matching and is set as per what Prisma's type is and then run again.

**Reply by TheoMer** [reply]
Hey @ryands17 I'm still having no joy with this. If I convert the following:

```
export type SizeCreateWithoutItemInput = {
  createdAt?: Date | string
  id: string
  label?: string | null
  name: string
  updatedAt?: Date | string
  itemvariants?: ItemVariantsCreateManyWithoutSizeInput | null
  orderitem?: OrderItemCreateManyWithoutSizeInput | null
}
```

to

```
schema.inputObjectType({
  name: 'SizeCreateWithoutItemInput',
  definition(t) {
    t.date('createdAt')
    t.string('id', { nullable: false })
    t.string('label')
    t.string('name', {nullable: false })
    t.date('updatedAt')
    t.field('itemvariants', { type: 'ItemVariantsCreateManyWithoutSizeInput' }) 
    t.field('orderitem', { type: 'OrderItemCreateManyWithoutSizeInput' })
  }
})
```

how do I specify label to have an output value of just 'string | null' and not 'string | null | undefined' ??? 

t.string('label', { nullable: true }) is still giving me 'string | null | undefined'.

**Reply by ryands17** [reply]
Hey @TheoMer 👋 
Let me reproduce with a part of your schema and get back to you on this :)

**Comment by TheoMer** [comment]
Hey @ryands17 that's great.

You can find the current iteration of my schema here: https://gist.github.com/TheoMer/219394ae817964d0c92200e73cfe48f5

**Reply by TheoMer** [reply]
Hey @ryands17 for some additional context I attach the following images showing my mutationType and typescript type it is based upon, and what the typescript type is expecting as valid values and what Size in the mutationType states it sees as valid values:


![prisma_property2](https://user-images.githubusercontent.com/15969262/90266214-8dbb5780-de4b-11ea-922b-0e7e1b747c6b.jpg)

![prisma_property](https://user-images.githubusercontent.com/15969262/90266206-898f3a00-de4b-11ea-8bc3-687c101519ad.jpg)


index.d.ts
```
export type ItemCreateInput = {
  createdAt?: Date | string
  description: string
  id: string
  image?: string | null
  image2?: string | null
  image3?: string | null
  image4?: string | null
  image5?: string | null
  image6?: string | null
  largeImage?: string | null
  largeImage2?: string | null
  largeImage3?: string | null
  largeImage4?: string | null
  largeImage5?: string | null
  largeImage6?: string | null
  mainDescription: string
  price: number
  quantity?: number
  title: string
  updatedAt?: Date | string
  userIdentity?: string
  Color: ColorCreateOneWithoutItemInput
  Size: SizeCreateOneWithoutItemInput
  User: UserCreateOneWithoutItemsInput
  cartitems?: CartItemCreateManyWithoutItemInput | null
  itemvariants?: ItemVariantsCreateManyWithoutItemInput | null
}
```

**Reply by ryands17** [reply, marked as answer]
Hey @TheoMer 👋 
I can reproduce this unfortunately is related to [this](https://github.com/graphql-nexus/nexus/issues/1058#issuecomment-645248108) issue.

The link above is a workaround that you would need to perform for the timebeing by explicitly adding the non-null fields instead of passing the entire data, or you would need to use `crud` to automatically create that resovler for you,

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
